arangod/Graph/GraphOperations.h

```diff
@@ -0,0 +1,202 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz & Heiko Kernbach
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGOD_GRAPH_GRAPHOPERATIONS_H
+#define ARANGOD_GRAPH_GRAPHOPERATIONS_H
+
+#include <velocypack/Buffer.h>
+#include <chrono>
+#include <utility>
+
+#include "Aql/Query.h"
+#include "Aql/VariableGenerator.h"
+#include "Basics/ReadWriteLock.h"
+#include "Cluster/ClusterInfo.h"
+#include "Cluster/ResultT.h"
+#include "Graph/Graph.h"
+#include "Transaction/Methods.h"
+#include "Transaction/StandaloneContext.h"
+#include "Utils/OperationResult.h"
+
+namespace arangodb {
+namespace graph {
+
+class GraphOperations {
+ private:
+  Graph const& _graph;
+  std::shared_ptr<transaction::Context> _ctx;
+
+  Graph const& graph() const { return _graph; };
+  std::shared_ptr<transaction::Context>& ctx() { return _ctx; };
+
+  /// @brief edge definitions of this graph
+  std::unordered_map<std::string, EdgeDefinition> _edgeDefs;
+
+ public:
+  GraphOperations() = delete;
+  GraphOperations(Graph const& graph_,
+                  std::shared_ptr<transaction::Context> ctx_)
+      : _graph(graph_), _ctx(std::move(ctx_)) {}
+
+  // TODO I added the complex result type for the get* methods to exactly
+  // reproduce (in the RestGraphHandler) the behaviour of the similar methods
+  // in the RestDocumentHandler. A simpler type, e.g. ResultT<OperationResult>,
+  // would be preferable.
+  /// @brief Graph collection name
+  static std::string const _graphs;
```

Comment:
What is this static constant for? Is there any value assigned to it somewhere?
And it seems redudant, because there is now a constant string for it in StaticStrings.

Reply:
resolved